### PR TITLE
chain: add `TestMempoolAccept` to chain interface

### PR DESCRIPTION
**Pull request by yyforyongyu**
`TestMempoolAccept` is now added to chain interface so it can be accessed via chain io in `lnd`.

### CODE REVIEW

File: wallet/wallet.go [hunk 1]
```diff
@@ -4131,7 +4131,7 @@ func MapBroadcastBackendError(err error) error {
 	//
 	// https://github.com/bitcoin/bitcoin/blob/9bf5768dd628b3a7c30dd42b5ed477a92c4d3540/src/node/transaction.cpp#L49
 	// https://github.com/bitcoin/bitcoin/blob/0.20/src/validation.cpp#L642
-	case match(err, "missing inputs") ||
+	case match(err, "missing-inputs") ||
```
Comment (Roasbeef) [inline wallet/wallet.go:4134]: 😣

Comment (Roasbeef) [reply wallet/wallet.go:4134]: Wondering what we can do in the future to catch stuff like this...

Ofc it's the case that string error matching is fragile to being with, wish we had some concrete error codes here.

Comment (Roasbeef) [reply wallet/wallet.go:4134]: Or should we be looking at one of these sources instead?
  * https://github.com/bitcoin/bitcoin/blob/4baa162dbb3c6464e196a4a21fe63794859021b4/src/util/error.cpp#L19
  * https://github.com/bitcoin/bitcoin/blob/4baa162dbb3c6464e196a4a21fe63794859021b4/src/consensus/tx_verify.cpp#L172

Comment (Roasbeef) [reply wallet/wallet.go:4134]: Also fwiw, this gives a 404 now: 
> https://github.com/bitcoin/bitcoin/blob/0.20/src/validation.cpp#L642

Comment (yyforyongyu) [reply wallet/wallet.go]: These two are my primary source when comparing RPC errors, as I'm not sure how `bitcoind` converts them internally, so I can only rely on the end results from its rpc responses,
- https://github.com/bitcoin/bitcoin/blob/master/test/functional/data/invalid_txs.py
- https://github.com/bitcoin/bitcoin/blob/master/test/functional/mempool_accept.py

Also updated the link, and added a new method to properly handle error matching. Meanwhile plan to work on something in `btcd/btcjson` - this way we can have a single place to track the bitcoind errors and handles the matching then turning them into defined error types.

**Comment by saubyk** [comment]
cc: @ellemouton

File: go.mod [hunk 2]
```diff
@@ -1,11 +1,11 @@
 module github.com/btcsuite/btcwallet
 
 require (
-	github.com/btcsuite/btcd v0.23.5-0.20230711222809-7faa9b266231
+	github.com/btcsuite/btcd v0.23.5-0.20231215221805-96c9fd8078fd
```
Comment (ellemouton) [inline go.mod:4]: think we can update now (unless you wanna wait for a tag on that side)

**Review by ellemouton — APPROVED** [review]
lgtm!

File: go.mod (hunk 2 above)
Comment (yyforyongyu) [reply go.mod]: cool updated!

**Review by Roasbeef — APPROVED** [review]
LGTM 🍭

**Pull request merged**